compiler/rustc_interface/src/interface.rs

```diff
@@ -59,9 +60,25 @@ impl Compiler {
     }
 }
 
+fn registry_setup() {
+    thread_local! {
+        static ONCE: OnceCell<()> = OnceCell::new();
+    }
+
+    // Create a dummy registry to allow `WorkerLocal` construction.
+    // We use `OnceCell` so we only register one dummy registry per thread.
```

Comment:
This is kind of hacky due to https://github.com/rust-lang/rust/pull/101313 now using `WorkerLocal` outside the Rayon thread pool.

Reply:
In this case we shouldn't use `WorkerLocal` in #101313. We could use `Cell<u32>` in single-thread mode and `AtomicU32` in parallel mode, but this require the `DynSendSyncCheck` trait I metioned in #107586.

Reply:
In this case #101313 shouldn't use `WorkerLocal`, since `WorkerLocal::new()` will reset the AttrIdAllocator. It can use `Cell<u32>` in single-thread mode and `AtomicU32` in parallel mode, but this require the `DynSendSync` I metioned in #107586

Reply:
Atomic adds are very cheap, reverting it may even be a performance improvement, so I think that's the best option.

Reply:
Actually changing the `PR` to use `AttrIdGenerator(AtomicU32)` is nicer, as it avoids the global which is kind of incorrect.

Reply:
Makes sense. But I think there will be some regression under single thread.